assets/js/state/sagas/hosts.js

```diff
@@ -1,7 +1,54 @@
-import { put, takeEvery } from 'redux-saga/effects';
-import { HOST_DEREGISTERED, removeHost } from '@state/hosts';
+import {
+  all,
+  delay,
+  put,
+  race,
+  call,
+  take,
+  takeEvery,
+} from 'redux-saga/effects';
+import {
+  DEREGISTER_HOST,
+  CANCEL_DEREGISTER_HOST,
+  HOST_DEREGISTERED,
+  removeHost,
+  setHostDeregisterable,
+} from '@state/hosts';
 import { notify } from '@state/actions/notifications';
 
+export function* markDeregisterableHosts(hosts) {
+  yield all(
+    hosts
+      .filter(({ heartbeat }) => heartbeat !== 'passing')
+      .map((host) =>
+        put({
+          type: DEREGISTER_HOST,
+          payload: host,
+        })
+      )
+  );
+}
+
+function* deregisterHostDetail(payload) {
+  // eslint-disable-next-line no-undef
+  yield delay(config.deregistrationDebounce ?? 0);
```

Comment:
Let's pass this `config` value as argument. The purest, the better.
We could get the value in the main sagas index file.
We will need to pass it through multiple functions, but it would be much better, to make things testable either

Reply:
+1